modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -308,6 +308,26 @@ namespace cv {
                     layer_id++;
                     fused_layer_names.push_back(last_layer);
                 }
+                
+                void setCrop(int crop_height, int crop_width)
+                {
+                    cv::dnn::LayerParams crop_param;
+                    crop_param.set<int>("crop_height", crop_height);
+                    crop_param.set<int>("crop_width", crop_width);
+                    crop_param.name = "CropLayer-name";
+                    crop_param.type = "CropLayer";
```

Comment:
There is no "CropLayer" in OpenCV - just "Crop"

Reply:
Moreover, there is no `crop_height` and `crop_width` attributes for Crop. You need to utilize Slice with it's `begin`, `end` (https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/layers/slice_layer.cpp):

```cpp
std::vector<int> begin = {0, 0, 0, 0};
std::vector<int> sizes = {-1, -1, crop_height, crop_width};
crop_param.set("begin", DictValue::arrayInt(&begin[0], begin.size()));
crop_param.set("size", DictValue::arrayInt(&sizes[0], sizes.size()));
crop_param.type = "Slice";
```

Reply:
if `crop_height` and `crop_width` are not attributes for crop. Why did we generate test data using those parameters in `opencv_extra`